### PR TITLE
AEA-1212 Fix for aea scaffold connection:

### PR DESCRIPTION
**Pull request by solarw**
Fix for aea scaffold connection: **kwargs for Connection.__init__ and public id fix for python files

## Fixes

https://github.com/fetchai/agents-aea/issues/2123

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/cli/scaffold.py [hunk 1]
```diff
@@ -187,11 +186,13 @@ def scaffold_item(ctx: Context, item_type: str, item_name: str) -> None:
...
             loader.dump(config, fp)
 
-        # update 'PUBLIC_ID' variable with the right public id
-        init_module = Path(dest, "__init__.py")
-        init_module.write_text(
-            re.sub(SCAFFOLD_PUBLIC_ID, str(new_public_id), init_module.read_text())
-        )
+        # update 'PUBLIC_ID' variable with the right public id in all python files!
+
+        for file_path in glob.glob(str(Path(dest, "*.py"))):
```
Comment (DavidMinarsch) [inline aea/cli/scaffold.py:191]: We're not looking for any file. We're looking for init file only..

File: aea/connections/scaffold/connection.py [hunk 2]
```diff
@@ -33,22 +31,15 @@ class MyScaffoldConnection(Connection):
...
         Initialize a connection to an SDK or API.
 
         :param configuration: the connection configuration.
         :param crypto_store: object to access the connection crypto objects.
         :param identity: the identity object.
         """
-        super().__init__(  # pragma: no cover
-            configuration=configuration, crypto_store=crypto_store, identity=identity
-        )
+        super().__init__(**kwargs)  # pragma: no cover
```
Comment (DavidMinarsch) [inline aea/connections/scaffold/connection.py:42]: The reason we did miss this before is because we had no test. Pragma no cover needs removing and we need to test instantiation of an empty scaffolded class.

Comment (DavidMinarsch) [reply aea/connections/scaffold/connection.py:42]: I see test is there now, so remove pragma statement please

File: aea/connections/scaffold/connection.py [hunk 3]
```diff
@@ -33,22 +31,15 @@ class MyScaffoldConnection(Connection):
...
-        identity: Optional[Identity] = None,
-        crypto_store: Optional[CryptoStore] = None,
-    ):
+    def __init__(self, **kwargs):
         """
         Initialize a connection to an SDK or API.
 
         :param configuration: the connection configuration.
         :param crypto_store: object to access the connection crypto objects.
         :param identity: the identity object.
```
Comment (DavidMinarsch) [inline aea/connections/scaffold/connection.py:40]: The docstring is no longer accurate.

File: aea/connections/scaffold/connection.py (hunk 2 above)
Comment (solarw) [reply aea/connections/scaffold/connection.py:42]: the problem that file is not executed, but it's copy, this is why pragma: nocover is used

File: aea/connections/scaffold/connection.py (hunk 3 above)
Comment (solarw) [reply aea/connections/scaffold/connection.py:40]: updated to the base connection definition

File: aea/cli/scaffold.py (hunk 1 above)
Comment (solarw) [reply aea/cli/scaffold.py:191]: now fixes only for `__init__.py` and `connection.py`

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Review by marcofavorito — APPROVED** [review]
LGTM
